test/layering_test.dart

```diff
@@ -0,0 +1,48 @@
+//  Copyright 2023 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import 'package:layerlens/src/analyzer.dart';
+import 'package:layerlens/src/model.dart';
+import 'package:test/test.dart';
+
+import 'test_infra/tests.dart';
+
+final _tests = <String, Dependencies>{
+  'simple': {
+    'a/b/c/d.dart': {},
+    'a/b/c.dart': {},
+    'a/b.dart': {},
+    'a.dart': {}
+  },
+};
+
+void main() {
+  test('creates folder structure', () async {
+    for (final name in _tests.keys) {
```

Comment:
Do the `for` outside of the `test` function, so you can add the `name` of the test as part of the name of the test, so the failure can be identified faster, something like:

```dart
for (final name in _test.keys) {
  test('creates folder structure - $name', () async {
  }
  ...
}
```

Or alternatively, use the `reason` parameter of the `expect` function to describe the expected result, and add the `name` of the test as part of the description.

Reply:
Test inside loop makes it harder for IDE to support the testing. Added reason.